---
bool 	is_ground_clear 	# whether ground is clear